exec/vector/src/main/java/org/apache/drill/exec/record/metadata/MetadataUtils.java

```diff
@@ -205,4 +205,10 @@ private static ColumnMetadata newDecimal(String name, MinorType type, DataMode m
     return new PrimitiveColumnMetadata(field);
   }
 
+  private static boolean isScalar(MinorType type) {
+    return type != MinorType.MAP
+        && type != MinorType.UNION
+        && type != MinorType.LIST
+        && type != MinorType.DICT;
+  }
```

Comment:
Maybe change this to `isComplex`, and define `isScalar()` as `! isComplex()`.

Reply:
Yeah, much negation... Changed the body of the method - didn't want to introduce the new `isComplex()`.